### ISSUE TITLE
Why are actions async by default?

### ISSUE THREAD

**Issue by n8agrin** [reflux/discuss#6]
Answered my own question!

**Comment by devinivy** [comment]
@n8agrin I realize this is old, but do you mind sharing what you found?  I was also curious if you were referring to `trigger()`/`triggerAsync()` or the `asyncResult` option.

**Comment by n8agrin** [comment]
@devinivy we use a somewhat bizarre implementation of Flux / Reflux in that our view layer is still powered by an aging Angular app.

Angular has its own digest loop, which acts similarly to Reflux' async actions. Both are attempting to define a release point for the browser to exit the JS thread (effectively using setTimeout calls under the hood), allowing the browser to update the DOM before executing the next scheduled chunk of Javascript work.

The day I created this comment I was very confused because we were seeing strange behavior that vanished when we converted our Reflux actions to act synchronously. I finally remembered that I had to also consider the nature of the underlying Angular app, which also defers events. Having two asynchronous systems on top of one another proved nearly impossible to reason about.
